### PR TITLE
Add Damask Upgrade docs

### PR DESCRIPTION
**Pull request by tjanez**

### CODE REVIEW

File: docs/general/mainnet/damask-upgrade.md [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+  https://oasiscommunity.slack.com/archives/CMUSJCRFA/p1647881564057319?thread_ts=1647448573.197229&cid=CMUSJCRFA
+
+## Mechanics of the Upgrade
+
+On Mar 24, 2022, the Oasis Protocol Foundation submitted the upgrade governance
+proposal with id of `2` which proposed upgrading the network at epoch 13402.
+
+:::tip
+
+As well as submitting the actual governance proposal to the network, Oasis
```
Comment (peterjgilbert) [inline docs/general/mainnet/damask-upgrade.md]:
```suggestion
In addition to submitting the actual governance proposal to the network, Oasis
```

File: docs/general/mainnet/damask-upgrade.md [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+Protocol Foundation also published the [Damask Upgrade Proposal discussion] to
+the [Oasis Community Forum on GitHub].
+
+:::
+
+Node operators which had an active validator node in the validator set had 1
+week to cast their vote.
+
+Validators representing more than 88% of the total staking power on the Oasis
+Network participated in the vote and 100% of them voted _yes_ for the proposal.
```
Comment (peterjgilbert) [inline docs/general/mainnet/damask-upgrade.md]:
```suggestion
Validators representing more than 88% of the total stake in the consensus
committee participated in the vote, and 100% of them voted _yes_ for the proposal.
```

File: docs/general/mainnet/damask-upgrade.md [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+
+This upgrade will be performed by exporting the network's state at the upgrade
```
Comment (peterjgilbert) [inline docs/general/mainnet/damask-upgrade.md]:
```suggestion
The upgrade will be performed by exporting the network's state at the upgrade
```

File: docs/general/mainnet/damask-upgrade.md [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+epoch, updating the [genesis document][genesis-doc], upgrading the Oasis Node
+and the ParaTime binaries and starting a new network from the new genesis file.
+
+This will require coordination between node operators and the Oasis Protocol
+Foundation.
+All nodes will need to configure the new genesis file that they can generate or
+verify independently and reset/archive any state from Mainnet.
```
Comment (peterjgilbert) [inline docs/general/mainnet/damask-upgrade.md]:
```suggestion
verify independently and reset/archive any existing state from Mainnet.
```

File: docs/general/mainnet/damask-upgrade.md [hunk 5]
```diff
@@ -0,0 +1,349 @@
...
+
+Once enough (representing 2/3+ of stake) nodes have taken this step, the
```
Comment (peterjgilbert) [inline docs/general/mainnet/damask-upgrade.md]:
```suggestion
Once enough nodes (representing 2/3+ of stake) have taken this step, the
```

File: docs/general/mainnet/damask-upgrade.md [hunk 6]
```diff
@@ -0,0 +1,349 @@
...
+  minimum number of tokens one can transfer.
+  It will be set to 10,000,000 base units, or 0.01 ROSE tokens.
+
+  This will be done automatically with the `oasis-node debug fix-genesis`
+  command.
+
+* **`staking.params.min_transact_balance`** is a new parameter that specifies
+  the minimum general balance an account must have to be able to perform
+  transactions on the network.
+  It will be set to 0 base units meaning this requirement is currently not
```
Comment (peterjgilbert) [inline docs/general/mainnet/damask-upgrade.md]:
```suggestion
  It will be set to 0 base units, meaning this requirement is currently not
```

**Review by peterjgilbert — APPROVED** [review]

File: docs/general/mainnet/damask-upgrade.md [hunk 7]
```diff
@@ -0,0 +1,349 @@
+# Damask Upgrade
+
+This document provides an overview of the changes for the Damask Mainnet
+upgrade.
+
+:::caution
+
+The Damask upgrade on Mainnet will kick off around **Apr 11, 2022 at 9:00 UTC**.
```
Comment (kostko) [inline docs/general/mainnet/damask-upgrade.md]: We should probably specify the epoch as well?

File: docs/general/mainnet/damask-upgrade.md [hunk 8]
```diff
@@ -0,0 +1,349 @@
...
+
+### **General**
+
+* **`height`** will be set to the height of the Mainnet state dump + 1, _TBD_.
+
+* **`genesis_time`** will be set to`2022-04-11T10:00:00Z`.
+
+* **`chain_id`** will be set to `oasis-3`.
+
+* **`halt_epoch`** will be set to`22500`(approximately 1 year from the Damask
```
Comment (kostko) [inline docs/general/mainnet/damask-upgrade.md]: I would just bump it by 10,000.

File: docs/general/mainnet/damask-upgrade.md [hunk 9]
```diff
@@ -0,0 +1,349 @@
...
+
+* **`beacon.params.backend`** configures the random beacon backend to use.
+  It will be set to `"vrf"` indicating that the beacon implementing
+  [VRF-based random beacon][ADR 0010] should be used.
+
+  This will be done automatically with the `oasis-node debug fix-genesis`
+  command.
+
+* **`beacon.params.vrf_parameters.alpha_hq_threshold`** is _TODO_.
+  It will be set to `2`.
```
Comment (kostko) [inline docs/general/mainnet/damask-upgrade.md]:
```suggestion
  It will be set to `20`.
```

File: docs/general/mainnet/damask-upgrade.md [hunk 10]
```diff
@@ -0,0 +1,349 @@
...
+
+  This will be done automatically with the `oasis-node debug fix-genesis`
+  command.
+
+* **`beacon.params.vrf_parameters.interval`** is _TODO_.
+  It will be set to `100`.
```
Comment (kostko) [inline docs/general/mainnet/damask-upgrade.md]:
```suggestion
  It will be set to `600`.
```

File: docs/general/mainnet/damask-upgrade.md [hunk 11]
```diff
@@ -0,0 +1,349 @@
...
+
+  This will be done automatically with the `oasis-node debug fix-genesis`
+  command.
+
+* **`beacon.params.vrf_parameters.proof_delay`** is _TODO_.
+  It will be set to `50`.
```
Comment (kostko) [inline docs/general/mainnet/damask-upgrade.md]:
```suggestion
  It will be set to `400`.
```

File: docs/general/mainnet/damask-upgrade.md (hunk 7 above)
Comment (tjanez) [reply docs/general/mainnet/damask-upgrade.md]: Done.

File: docs/general/mainnet/damask-upgrade.md (hunk 9 above)
Comment (tjanez) [reply docs/general/mainnet/damask-upgrade.md]: Done.

File: docs/general/mainnet/damask-upgrade.md (hunk 10 above)
Comment (tjanez) [reply docs/general/mainnet/damask-upgrade.md]: Done.

File: docs/general/mainnet/damask-upgrade.md (hunk 11 above)
Comment (tjanez) [reply docs/general/mainnet/damask-upgrade.md]: Done.

File: docs/general/run-a-node/upgrade-log.md [hunk 12]
```diff
@@ -167,17 +167,213 @@ oasis-node consensus submit_tx \
...
+
+[22.1]: https://github.com/oasisprotocol/oasis-core/releases/tag/v22.1
 
 ### Instructions - Upgrade day
 
-_TBD_
+Following steps should be performed on **2022-04-11** only after the network has
+reached the upgrade epoch and has halted:
+
+* Download the genesis file published in the [Damask Upgrade release].
```
Comment (matevz) [inline docs/general/run-a-node/upgrade-log.md]: If the steps below are order-sensitive, I suggest we use numbering:
```suggestion
1. Download the genesis file published in the [Damask Upgrade release].
```

File: docs/general/run-a-node/upgrade-log.md [hunk 13]
```diff
@@ -167,17 +167,213 @@ oasis-node consensus submit_tx \
...
+
+Mainnet state at epoch **13402** will be exported and migrated to a 22.1.x
+compatible genesis file.
+
+Upgrade genesis file will be published on the above link soon after reaching the
+upgrade epoch.
+
+:::
+
+* Verify the provided Damask Upgrade genesis file by comparing it to network
```
Comment (matevz) [inline docs/general/run-a-node/upgrade-log.md]: Same here and below.

File: docs/general/mainnet/damask-upgrade.md (hunk 8 above)
Comment (tjanez) [reply docs/general/mainnet/damask-upgrade.md]: Updated.

File: docs/general/run-a-node/upgrade-log.md (hunk 12 above)
Comment (tjanez) [reply docs/general/run-a-node/upgrade-log.md]: Thanks for the suggestion. Updated.

**Pull request merged**